web/service-worker.js

```diff
@@ -0,0 +1,36 @@
+function openWindowWithUrl (targetUrl, clientWindows) {
+  for (let i = 0, count = clientWindows.length; i < count; ++i) {
+    if (clientWindows[i].url === targetUrl && 'focus' in clientWindows[i]) {
+      clientWindows[i].focus();
+      return clientWindows[i];
+    }
+  }
+
+  // If site is cannot be opened, open in new window with url
+  if (self.clients.openWindow) {
+    self.clients.openWindow(targetUrl);
+  }
+
+  return null;
+}
+
+self.addEventListener('notificationclick', (event) => {
+  event.notification.close();
+  const notificationUrl = event?.notification?.data?.url;
```

Comment:
Believe it or not, Firefox for Android supports service workers, but it will break because of the `?.` operators!

Reply:
ah shit. I thought I got them all. dangit